src/api/java/mekanism/api/SerializerHelper.java

```diff
@@ -48,6 +49,81 @@ private SerializerHelper() {
         return Codec.LONG.flatXmap(checker, checker);
     });
 
+    @Deprecated(since = "10.6.6", forRemoval = true)//TODO - 1.22: Remove
+    private static final Codec<Long> LEGACY_CODEC_FLOATING_LONG = new PrimitiveCodec<>() {
+        @Override
+        public <T> DataResult<Long> read(DynamicOps<T> ops, T input) {
+            return ops.getStringValue(input).flatMap(number -> {
+                try {
+                    long value;
+                    int index = number.indexOf('.');
+                    if (index == -1) {
+                        value = Long.parseUnsignedLong(number);
+                    } else {
+                        value = Long.parseUnsignedLong(number.substring(0, index));
+                    }
+                    if (value < 0) {
+                        //Clamp unsigned to positive.
+                        value = Long.MAX_VALUE;
+                    }
+                    if (value == 0) {
+                        //If we are at zero, see if we should ceil the decimal
+                        if (index == -1) {
+                            return DataResult.success(0L);
+                        }
+                        String decimalAsString = number.substring(index + 1);
+                        int numberDigits = decimalAsString.length();
+                        if (numberDigits < 4) {
```

Comment:
Since this is not using the result for anything but a `> 0` check, does it really need to be padded? We could just parse as long and get the same result